lib/macho/load_commands.rb

```diff
@@ -268,7 +268,8 @@ def initialize(lc, lc_str)
         if view
           lc_str_abs = view.offset + lc_str
           lc_end = view.offset + lc.cmdsize - 1
-          @string = view.raw_data.slice(lc_str_abs..lc_end).delete("\x00")
+          raw_string = view.raw_data.slice(lc_str_abs..lc_end)
+          @string = raw_string.split("\x00").first
```

Comment:
I made the condition `@string.empty? || null_byte.empty?`, since we probably also want to prevent the user from doing something like `change_install_name(old, "\x00")`.